src/file_reader.cpp

```diff
@@ -143,7 +143,7 @@ class MMapException : std::exception {};
 char*
 mmapReadOnly(int fd, offset_type offset, size_type size)
 {
-#if defined(__APPLE__) || defined(__OpenBSD__) || defined(__HAIKU__)
+#if defined(__APPLE__) || defined(__NetBSD__) || defined(__OpenBSD__) || defined(__HAIKU__)
```

Comment:
Would it be better to add `MAP_POPULATE` only if on Linux instead of excluding new (and yet to know) platform where it is not available ?

Reply:
Yes. It would be better to chaneg to:

```
#if defined(__linux__)
#endif
```

MAP_POPULATE is Linux-specific

Reply:
```diff
diff --git a/src/file_reader.cpp b/src/file_reader.cpp
index 0efaeae..c5ef8e5 100644
--- a/src/file_reader.cpp
+++ b/src/file_reader.cpp
@@ -143,12 +143,12 @@ class MMapException : std::exception {};
 char*
 mmapReadOnly(int fd, offset_type offset, size_type size)
 {
-#if defined(__APPLE__) || defined(__OpenBSD__) || defined(__HAIKU__)
-  const auto MAP_FLAGS = MAP_PRIVATE;
+#elif defined(__linux__)
+  const auto MAP_FLAGS = MAP_PRIVATE|MAP_POPULATE;
 #elif defined(__FreeBSD__)
   const auto MAP_FLAGS = MAP_PRIVATE|MAP_PREFAULT_READ;
 #else
-  const auto MAP_FLAGS = MAP_PRIVATE|MAP_POPULATE;
+  const auto MAP_FLAGS = MAP_PRIVATE;
 #endif
 
   const auto p = (char*)mmap(NULL, size, PROT_READ, MAP_FLAGS, fd, offset);

```